driver/ni/scanner.cpp

```diff
@@ -84,12 +79,8 @@ void ni::Scanner::scan() {
         this->filter, NULL,
         &this->resources_handle
     );
-    if (err != NISysCfg_OK) {
-        this->ok_state = false;
-        LOG(ERROR) << "[ni.scanner] failed to find hardware for task " << this->task.
-                name;
-        return;
-    }
+    if (err != NISysCfg_OK) return log_err("failed to find hardware");
```

Comment:
Again, is it ok that we got rid of the early return?

Reply:
this one still returns